### PR TITLE
[T4A4][F11-B1] Justin Ng

### PR DESCRIPTION
**Pull request by njyjn**
Ready for review.

### CODE REVIEW

**Comment by njyjn** [comment]
@Candiie Regarding what was discussed in tutorial, I have spelled out the method in the interface since the overidden methods are all the same. Is there a way to access the variables within the class implementing an interface, as shown in the example below?

```
    @Override
    public String getPrintableString() {
        return getClass().getName() + ": " + fullName;
    }
```

to

```
public interface Printable {
    /**
     * Returns a concatenated version of the printable strings of each object.
     */
    public default String getPrintableString() {
        return getClass().getName() + ": " + value;
    }
}
```

**Comment by Candiie** [comment]
@njyjn  Hi Justin, actually I'm not exactly sure what you mean by "Is there a way to access the variables within the class implementing an interface", maybe you can email me @ cwbay@u.nus.edu for further clarification (try to include a context where you might need to access the variables). Otherwise, if its too difficult to explain through email, perhaps we can take a look at it in the next tutorial :)

File: src/seedu/addressbook/data/person/Email.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * Represents a Person's email in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidEmail(String)}
  */
-public class Email {
+public class Email implements Printable {
 
     public static final String EXAMPLE = "valid@e.mail";
     public static final String MESSAGE_EMAIL_CONSTRAINTS =
```
Comment (Candiie) [inline src/seedu/addressbook/data/person/Email.java:12]: Hi Justin, erm is that a missing chunk of code i'm looking at o.O? (after the "=" sign)

File: src/seedu/addressbook/ui/TextUi.java [hunk 2]
```diff
@@ -164,5 +165,21 @@ private static String getIndexedListForViewing(List<String> listItems) {
...
+    }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    String getPrintableString(Printable... printables) {
+    	final StringBuilder builder = new StringBuilder();
+    	for (Printable printable : printables) {
+    		builder.append(printable.getPrintableString());
+    		builder.append(" ");
```
Comment (Candiie) [inline src/seedu/addressbook/ui/TextUi.java:180]: I think this result in an extra " " at the end. Take a look @ StringJoiner or String.join as mentioned in tutorial :)

**Comment by Candiie** [comment]
Some comments added. Please close the PR after reading comments.

Hi Hi, great effort! Do take a look @ the comments above ~

File: src/seedu/addressbook/data/person/Printable.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package seedu.addressbook.data.person;
+
+/**
+ * A read-only immutable interface for a Person's field in the addressbook
```
Comment (Candiie) [inline src/seedu/addressbook/data/person/Printable.java:4]: :+1:  for header comment !!! :)

**Pull request closed without merging**